bin/check_deadlock.sh

```diff
@@ -0,0 +1,62 @@
+#!/bin/bash
+
+#config
+user=root
+passwd=
+fe_ip=127.0.0.1
+http_port=8030
+timeout=30
+max_timeout_times=5
+fe_bin_path=/home/disk1/sr/app/starrocks-2.3.3/fe-5963b718-fdc4-42fc-a83c-5256cdcef805/bin
+jstack_path=/home/disk1/sr/app/starrocks-2.3.3/fe-5963b718-fdc4-42fc-a83c-5256cdcef805
+
+
+timeout_times=0
+interval=60
+while true
+do
+    # call show proc '/statistic' to check db deadlock
+    start_time=$(date +%s)
+    curl -u ${user}:${passwd} http://${fe_ip}:${http_port}/api/show_proc?path=/statistic --max-time ${timeout}
+    succ=$?
+    end_time=$(date +%s)
+
+    # check result, if return failed and time used is at least ${timeout}, increase timeout_times
+    if [ ${succ} -ne 0 ]; then
```

Comment:
I don't think curl will give you a succ=0 if the http request is processed but the response code is not OK. for example
```
$ curl http://10.152.183.84:8030/api/show_proc?path=/abc ; echo $?
{"status":"FAILED","msg":"Need auth information."}0
```
should make curl with -f option so that non-OK response will be taken as failure as well.